src/gui/mzroll/tabledockwidget.cpp

```diff
@@ -1,2532 +1,2847 @@
-    connect(btnMergeMenu, SIGNAL(triggered(QAction*)), SLOT(mergeGroupsIntoPeakTable(QAction*)));
-
-
-
-
-    QWidgetAction *titlePeakTable = new TableToolBarWidgetAction(toolBar, this,  "titlePeakTable");
-    QWidgetAction *btnSwitchView = new TableToolBarWidgetAction(toolBar, this,  "btnSwitchView");
-    QWidgetAction *btnGroupCSV = new TableToolBarWidgetAction(toolBar, this, "btnGroupCSV");
-    QWidgetAction *btnSaveJson = new TableToolBarWidgetAction(toolBar, this,  "btnSaveJson");
-    QWidgetAction *btnScatter = new TableToolBarWidgetAction(toolBar, this,  "btnScatter");
-    QWidgetAction *btnCluster = new TableToolBarWidgetAction(toolBar, this,  "btnCluster");
-    QWidgetAction *btnTrain = new TableToolBarWidgetAction(toolBar, this,  "btnTrain");
-    QWidgetAction *btnXML = new TableToolBarWidgetAction(toolBar, this,  "btnXML");
-    QWidgetAction *btnGood = new TableToolBarWidgetAction(toolBar, this,  "btnGood");
-    QWidgetAction *btnBad = new TableToolBarWidgetAction(toolBar, this,  "btnBad");
-    QWidgetAction *btnHeatmapelete = new TableToolBarWidgetAction(toolBar, this,  "btnHeatmapelete");
-    QWidgetAction *btnPDF = new TableToolBarWidgetAction(toolBar, this,  "btnPDF");
-    QWidgetAction *btnX = new TableToolBarWidgetAction(toolBar, this,  "btnX");
-
-    QWidget* spacer = new QWidget();
-    spacer->setSizePolicy(QSizePolicy::MinimumExpanding, QSizePolicy::Minimum);
-
-    toolBar->addAction(titlePeakTable);
-    toolBar->addAction(btnSwitchView);
-    toolBar->addAction(btnGood);
-    toolBar->addAction(btnBad);
-    toolBar->addAction(btnTrain);
-    toolBar->addAction(btnHeatmapelete);
-    toolBar->addWidget(btnMerge);
-
-    toolBar->addSeparator();
-    toolBar->addAction(btnScatter);
-    toolBar->addAction(btnCluster);
+TableDockWidget::TableDockWidget(MainWindow *mw) {
+  setAllowedAreas(Qt::AllDockWidgetAreas);
+  setFloating(false);
+  _mainwindow = mw;
 
+  pal = palette();
+  setAutoFillBackground(true);
+  pal.setColor(QPalette::Background, QColor(170, 170, 170, 100));
+  setPalette(pal);
 
-    toolBar->addSeparator();
-    toolBar->addAction(btnPDF);
-    toolBar->addAction(btnGroupCSV);
-    toolBar->addAction(btnSaveJson);
+  viewType = groupView;
 
+  treeWidget = new QTreeWidget(this);
+  treeWidget->setSortingEnabled(false);
+  treeWidget->setDragDropMode(QAbstractItemView::DragOnly);
+  treeWidget->setSelectionMode(QAbstractItemView::ExtendedSelection);
+  treeWidget->setAcceptDrops(false);
+  treeWidget->setObjectName("PeakGroupTable");
+  treeWidget->setFocusPolicy(Qt::NoFocus);
+  treeWidget->setSelectionBehavior(QAbstractItemView::SelectRows);
+  this->setFocusPolicy(Qt::ClickFocus);
+  tableSelectionFlagUp = false;
+  tableSelectionFlagDown = false;
+  this->setAcceptDrops(true);
 
-    toolBar->addSeparator();
-    toolBar->addAction(btnXML);
+  setWidget(treeWidget);
+  setupPeakTable();
+  tableId = (_mainwindow->noOfPeakTables)++;
```

Comment:
@saifulbkhan this should only happen for peakTableDockWidget. Also check wherever this counter is being used and make sure it won't cause any issues.

Reply:
WIll do.